### ISSUE TITLE
Allow to download an entire Playlist

### ISSUE THREAD

**Issue by MatheoDumont** [deepjyoti30/ytmdl#125]
One could pass the playlist url and download it.

**Comment by deepjyoti30** [comment]
@MatheoDumont It does support downloading a playlist at the moment. You can pass the playlist in the following way and ytmdl will automatically detect it

```console
ytmdl https://www.youtube.com/playlist\?list\=<playlist_id>
```

If you're referring to some other feature please be a bit more specific.

**Comment by Donokami** [comment]
```
❯ ytmdl https://www.youtube.com/playlist\?list\=PL4xG-EyB2mGsoXsBkjNhTRCyoRjwS19gU
 ==>  Youtube playlist passed...extracting! 
 ==>  Something went wrong while extracting the playlist data. 
Traceback (most recent call last):
  File "/home/donokami/.local/bin/ytmdl", line 469, in <module>
    extract_data()
  File "/home/donokami/.local/bin/ytmdl", line 439, in extract_data
    songs, playlist_name = yt.get_playlist(
TypeError: cannot unpack non-iterable NoneType object
```
I do have this issue when trying to dl a playlist

**Comment by deepjyoti30** [comment]
@Donokami Thanks for pointing it out. This was a bug in the code. However, it seems like you're not able to access this playlists. I just tried this playlists myself and it is working all right.

Are you able to extract other playlists? Are you able to extract this playlist using `youtube-dl`?

**Comment by Donokami** [comment]
> @Donokami Thanks for pointing it out. This was a bug in the code. However, it seems like you're not able to access this playlists. I just tried this playlists myself and it is working all right.
> 
> Are you able to extract other playlists? Are you able to extract this playlist using `youtube-dl`?

Yep it was a `youtube-dl` issue, but I still had some issue with https; but it is more linked to #138

**Comment by deepjyoti30** [comment]
@Donokami #138 is fixed in 48200d9fdf22e2fa009c74b623bd2bd84008fcc7

**Comment by DocMAX** [comment]
I have similar issue:
```
 ==> https://www.youtube.com/watch?v=C7-kAfQkZ9A: is unauthorized 
Traceback (most recent call last):
  File "/usr/bin/ytmdl", line 562, in <module>
    extract_data()
  File "/usr/bin/ytmdl", line 555, in extract_data
    main(args)
  File "/usr/bin/ytmdl", line 264, in main
    link, yt_title = search(song_name=song_name, args=args)
TypeError: cannot unpack non-iterable NoneType object

```
